src/utils/containers.ts

```diff
@@ -1,22 +1,65 @@
 import { strict as assert } from "assert";
 import { critical_error, M } from "./debugging-and-logging.js";
-import { set_interval, clear_interval, clear_timeout, set_timeout } from "./node.js";
+import { set_interval, clear_interval, set_timeout, clear_timeout } from "./node.js";
 
-export class SelfClearingSet<T> {
+export function force_clear_containers() {}
+
+let had_error = false;
+
+function set_clearing_interval(container: WeakRef<SelfClearingContainer>, interval_time: number) {
+    let cleared = false;
+    const interval: NodeJS.Timeout | null = set_interval(() => {
+        let container_ref = container.deref();
+        if (container_ref) {
+            container_ref.sweep();
+        } else if (interval !== null) {
+            clear_interval(interval);
+            cleared = true;
+        }
+        if (cleared && !had_error) {
+            // This is a warning because it's a sign of cleanup not happening properly, but it's not a critical error
+            // Also, it happens in tests, for some reason
+            M.warn(`Running cleared interval ${interval} at ${new Error().stack}`);
+            had_error = true;
+        }
```

Comment:
Is this logic still correct? If I'm understanding correctly it always runs the first time a container clears.

Reply:
It should never run, as `cleared` is already false when this runs for the first time, until `clear_interval` is called.
Though it still gets called in `vi.runPendingTimers()` for some reason (I say some reaon as the `clear_interval` runs and prints before this does)